lib/commander/command.rb

```diff
@@ -194,6 +194,16 @@ def proxy_option_struct
       end
     end
 
+    ##
+    # Do the same thing as `proxy_option_struct` but empty the proxy
+    # options.
+
+    def proxy_option_struct!
```

Comment:
It's been a while since I've looked at the `proxy_option_struct` code, but the behavior here seems surprising. I think it's probably less confusing to empty the array in `call`, perhaps with a note about why that's necessary.

Reply:
imho that whole strategy is a bit weird—mixing per-execution state with fields that last the life of the object—but i can see why it was done. I thought about just adding a line of code to `#call` but ended up making a `#proxy_option_struct!` alongside `#proxy_option_struct` since if anybody is using that API directly at all, that idiom (run `proxy_option_struct` and then `proxy_options.clear`) is going to pop up elsewhere. No sweat off my back to change it though.